packages/regenerator-transform/src/visit.js

```diff
@@ -16,6 +16,53 @@ import * as util from "./util";
 
 let getMarkInfo = require("private").makeAccessor();
 
+// we want to convert this shorthand object method into a normal object property
+// which is a function expression. for example, this:
+//  var foo = {
+//    *bar(baz) { return 5; }
+//  }
+// should be replaced with:
+//  var foo = {
+//    bar: function*(baz) { return 5; }
+//  }
+// to do this, we will clone the parameter array and the body of the function
+// into a new FunctionExpression.
+function replaceShorthandObjectMethod(path) {
+  if (!t.isObjectMethod(path.node)) {
+    return path;
+  }
+
+  // if it's not a generator method, we don't need to do anything.
+  if (!path.node.generator) {
+    return path;
+  }
+
+  const parameters = path.node.params.map(function (param) {
+    return t.cloneDeep(param);
+  })
+
+  if (!(path.node.key.name || path.node.key.value)) throw path.node.key;
+  const functionExpression = t.functionExpression(
+    null, // id
+    parameters, // params
+    t.cloneDeep(path.node.body), // body
+    path.node.generator,
+    path.node.async
+  );
+
+  path.replaceWith(
+    t.objectProperty(
```

Comment:
There is no existing `ObjectProperty`, there's only an `ObjectMethod`. 

When you have something like:

```
{ foo: function() {} };
```

the AST looks like this (with some parts emitted, obviously):

```
ObjectExpression {
  properties: [
    ObjectMethod {
      name: "foo",
      params: [],
      body: BlockStatement {
        body:[]
      }
    }
  ]
}
```

The point of this function is to replace the `ObjectMethod` with an `ObjectProperty`.

For more info, here's an AST Explorer that shows how `ObjectMethod` and `ObjectProperty` show up in the AST: https://astexplorer.net/#/gist/f4ba155b3c70312a3d84d100a5294417/ab381e4c3cc009f2a88e6aca29f0cc972974b81d